pkg/antctl/raw/packetcapture/command.go

```diff
@@ -0,0 +1,325 @@
+	}
+	if option.number == 0 {
+		return errors.New("packet number should be larger than 0")
+	}
+
+	_, _, antreaClient, err := getClients(cmd)
+	if err != nil {
+		return err
+	}
+	pc, err := newPacketCapture()
+	if err != nil {
+		return fmt.Errorf("error when filling up PacketCapture config: %w", err)
+	}
+	createCtx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	if _, err := antreaClient.CrdV1alpha1().PacketCaptures().Create(createCtx, pc, metav1.CreateOptions{}); err != nil {
+		return fmt.Errorf("error when creating PacketCapture, is PacketCapture feature gate enabled? %w", err)
+	}
+	defer func() {
+		if !option.nowait {
+			if err = antreaClient.CrdV1alpha1().PacketCaptures().Delete(context.TODO(), pc.Name, metav1.DeleteOptions{}); err != nil {
+				fmt.Fprintf(cmd.OutOrStdout(), "error when deleting PacketCapture: %s", err.Error())
+			}
+		}
+	}()
+
+	if option.nowait {
+		fmt.Fprintf(cmd.OutOrStdout(), "PacketCapture Name:  %s\n", pc.Name)
+		return nil
+	}
+
+	var latestPC *v1alpha1.PacketCapture
+	err = wait.PollUntilContextTimeout(context.TODO(), 1*time.Second, option.timeout, false, func(ctx context.Context) (bool, error) {
+		res, err := antreaClient.CrdV1alpha1().PacketCaptures().Get(ctx, pc.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		for _, cond := range res.Status.Conditions {
+			if cond.Type == v1alpha1.PacketCaptureComplete && cond.Status == metav1.ConditionTrue {
+				latestPC = res
+				return true, nil
+			}
+		}
+		return false, nil
+
+	})
+
+	if wait.Interrupted(err) {
+		err = errors.New("timeout waiting for PacketCapture done")
+		if latestPC == nil {
+			return err
+		}
+	} else if err != nil {
+		return fmt.Errorf("error when retrieving PacketCapture: %w", err)
+	}
+
+	splits := strings.Split(latestPC.Status.FilePath, ":")
+	fileName := filepath.Base(splits[1])
+	copier, _ := getCopier(cmd)
+	err = copier.CopyFromPod(context.TODO(), env.GetAntreaNamespace(), splits[0], "antrea-agent", splits[1], option.outputDir)
+	if err == nil {
+		fmt.Fprintf(cmd.OutOrStdout(), "Packet File: %s\n", filepath.Join(option.outputDir, fileName))
+	}
+	return err
+}
+
+func parseEndpoint(endpoint string) (pod *v1alpha1.PodReference, ip *string) {
+	parsedIP := net.ParseIP(endpoint)
+	if parsedIP != nil && parsedIP.To4() != nil {
+		ip = ptr.To(parsedIP.String())
+	} else {
+		split := strings.Split(endpoint, "/")
+		if len(split) == 1 {
+			pod = &v1alpha1.PodReference{
+				Namespace: "default",
+				Name:      split[0],
+			}
+		} else if len(split) == 2 && len(split[0]) != 0 && len(split[1]) != 0 {
+			pod = &v1alpha1.PodReference{
+				Namespace: split[0],
+				Name:      split[1],
+			}
+		}
+	}
+	return
+}
+
+func getPCName(src, dest string) string {
+	replace := func(s string) string {
+		return strings.ReplaceAll(s, "/", "-")
+	}
+	prefix := fmt.Sprintf("%s-%s", replace(src), replace(dest))
+	if option.nowait {
+		return prefix
+	}
```

Comment:
I don't get this, why to make the name differential with or without `nowait` since it's generated automatically?
btw, I suppose the name can also be an optional flag?

Reply:
nowait=true, means we will return directly after create the PC and left the user to check it out later(we don't delete this PC automatically also) , in this case, we want the PC name to be more human friendly.

If nowait=false , which is by default, we assume user only want to get the result and don't care the underlaying PC. so the name is irrelevant. This may help reduce the probability of name conflict.

This doesn't make much difference , i agree. Just copied this idea from Traceflow.